src/devtools/client/webconsole/components/Output/ConsoleOutput.tsx

```diff
@@ -160,44 +179,18 @@ class ConsoleOutput extends React.Component {
         });
       });
 
-    return React.createElement(
-      "div",
-      {
-        className: "webconsole-output",
-        role: "main",
-        ref: node => {
-          this.outputNode = node;
-        },
-      },
-      <ConsoleLoadingBar />,
-      messageNodes
+    return (
+      <div className="webconsole-output" ref={this.outputNode} role="main">
```

Comment:
maybe i'm being pedantic, but `ref={this.outputNode}` could be it's own PR as well and much easier to review and see if something broke...

Reply:
I'd classify it as a relatively low-risk change, especially given that the file is now TS.  All uses of `this.outputNode` now specifically use `this.outputNode.current!` and the types line up.  Plus, I would have had to define `outputNode = HTMLDivElement | null` anyway if I was going to stick with the callback form, so doing it as a ref object now simplifies that.